### PR TITLE
Localization

### PR DESCRIPTION
**Pull request by jau228**
This is basically pulling strings from xaml and place them in the resource file. And using static binding to link those strings. I also added an exception in the GitHubVS.ruleset to ignore CA1701 for the string renaming error("UserName" instead of "username", "Github" instead of "GitHub", etc...) I am not seeing a whole lot of strings beside the GitHub.VisualStudio project. So this is what I focus so far.

### CODE REVIEW

**Comment by shana** [comment]
Thanks for taking this on!

There are some strings on .cs files that need to be moved to resources too, particularly in the "TeamExplorer" folder.

File: src/common/GitHubVS.ruleset [hunk 1]
```diff
@@ -37,6 +37,7 @@
         <Rule Id="CA1501" Action="None" />
         <Rule Id="CA1502" Action="None" />
         <Rule Id="CA1506" Action="None" />
+		<Rule Id="CA1701" Action="None" />
```
Comment (Haacked) [inline src/common/GitHubVS.ruleset]: Minor nit, but could you fix the indenting? We use spaces, not tabs.

File: src/common/GitHubVS.ruleset [hunk 2]
```diff
@@ -67,6 +68,6 @@
         <Rule Id="CA2210" Action="None" />
         <Rule Id="CA2211" Action="None" />
         <Rule Id="CA2213" Action="None" />
-        <Rule Id="CA2227" Action="None" />
+        <Rule Id="CA2227" Action="None" />
```
Comment (Haacked) [inline src/common/GitHubVS.ruleset]: Please remove the extra whitespace at the end of this line.

**Comment by Haacked** [comment]
@jau228 THANK YOU for taking this on! It may take a little time for us to fully review. We've left you a few minor notes already.

**Comment by jau228** [comment]
I will move those strings from TeamExplorer folder this week.

**Comment by jau228** [comment]
The naming convention i used so far is {ControlName}{AttributeName}. eg. loginButtonContent, nameTextBlockText, helpCheckBoxToolTip, etc...

**Comment by Haacked** [comment]
> The naming convention i used so far is {ControlName}{AttributeName}. eg. loginButtonContent, nameTextBlockText, helpCheckBoxToolTip, etc...

That seems good to me. @shana?

**Comment by shana** [comment]
>> The naming convention i used so far is {ControlName}{AttributeName}. eg. loginButtonContent, nameTextBlockText, helpCheckBoxToolTip, etc...

> That seems good to me. @shana?

Yup, sounds good, that way it'll be easy to search&replace if we need to :+1:

**Comment by shana** [comment]
Hey hey! Just dropping a note to let you know this is not forgotten! I need to ship a release before I tackle open PRs (of which there are a few), I'll get to this soon!

**Comment by jau228** [comment]
yep, no problem. I am new to GitHub, so I am just trying to find projects to contribute.

**Pull request merged**